libvast/src/view.cpp

```diff
@@ -38,12 +33,22 @@ bool pattern_view::case_insensitive() const {
   return case_insensitive_;
 }
 
-bool operator==(pattern_view x, pattern_view y) noexcept {
-  return x.string() == y.string();
+bool operator==(pattern_view lhs, pattern_view rhs) noexcept {
+  return std::tie(lhs.pattern_, lhs.case_insensitive_)
+         == std::tie(rhs.pattern_, rhs.case_insensitive_);
 }
 
-bool operator<(pattern_view x, pattern_view y) noexcept {
-  return x.string() < y.string();
+std::strong_ordering operator<=>(pattern_view lhs, pattern_view rhs) noexcept {
+  while (!lhs.pattern_.empty() && !rhs.pattern_.empty()) {
+    if (lhs.pattern_[0] < rhs.pattern_[0])
+      return std::strong_ordering::less;
+    if (lhs.pattern_[0] > rhs.pattern_[0])
+      return std::strong_ordering::greater;
+    lhs.pattern_ = lhs.pattern_.substr(1);
+    rhs.pattern_ = rhs.pattern_.substr(1);
+  }
+  return static_cast<uint8_t>(lhs.case_insensitive_)
+         <=> static_cast<uint8_t>(rhs.case_insensitive_);
```

Comment:
This reads like magic to me. Can we add a comment why we need these gymnastics?

Reply:
This is just a polyfill for `std::lexicographical_compare_threeway`, which is not implemented in libc++. We should probably just add it somewhere to detail, as we use this in quite a few places now. Bools are not threeway-comparable, so we cast them to an 8-bit integer here.

Reply:
I added a comment in https://github.com/tenzir/vast/pull/2964/commits/6e6484023e289d13e9544ee5927577c002eacbb7